### ISSUE TITLE
nextDayThreshold should affect parsing of allDay event dates

### ISSUE THREAD

**Issue by bennyfio-ebi** [fullcalendar/fullcalendar#2985]
nextDayThreshold property is not applied to allDay events. 

For example see this jsfiddle:
http://jsfiddle.net/hL68co89/3/

I would expect the 2 all day event to be visible on day 17 of September like the non all day event, otherwise there is no visibility that the event is planned to last on 17 too.
The other worse side effect is on Day view, as you can see in the jsfiddle if you switch to day view, the two events are not visible at all.

nextDayThreshold should be applied consistently, independently from allDay, as the documentation says:
http://fullcalendar.io/docs/event_rendering/nextDayThreshold/
"Only affects timed events that appear on whole-days. Whole-day cells occur in month view, basicDay, basicWeek and the all-day slots in the agenda views." 
or in alternative an option should be provided to have the same effect on allDay events.

**Comment by arshaw** [comment]
when `allDay` is `true`, the time part of the ISO8601 is ignored. it does not consider `nextDayThreshold` by design. `nextDayThreshold` is used to determine how to display timed events that span into a new day.

However, this question has been brought up a number of times. It might be a good idea to respect `nextDayThreshold` in this case. I will keep this issue open for discussion.

**Comment by bennyfio-ebi** [comment]
Thanks for the reply. The reason for the number of times being questioned is that with current behavior it's easy for the agenda week and agenda day views to become inconsistent each other. You can get in agenda week view a day where the event is not visible anymore in all-day and is not visible in the week day column either, so you completely miss visibility that something is still going on, but then you switch to day view for that empty week day and you will see that there is an event still running from the day before.

**Comment by arshaw** [comment]
@bennyfio-ebi would you mind making a [JSBin example](http://fullcalendar.io/support/) of what you are talking about. i sort of get it, but would be helpful

**Comment by bennyfio-ebi** [comment]
@arshaw from my first message:
"
http://jsfiddle.net/hL68co89/3/

I would expect the 2 all day event to be visible on day 17 of September like the non all day event, otherwise there is no visibility that the event is planned to last on 17 too.
The other worse side effect is on Day view, as you can see in the jsfiddle if you switch to day view, the two events are not visible at all.
"

**Comment by arshaw** [comment]
enough people have been confused by this where I feel it's now a good idea to run the timed events through nextDayThreshold before converting them to [ambiguously-timed moments](http://fullcalendar.io/docs/utilities/Moment/#ambiguously-timed) for allDay events

**Comment by HartLarsson** [comment]
I've folowed the discussion and tested in tyhe demo that soemthing not working.
Considering the record:

```
				{
					title: 'Long Event',
					start: '2016-06-07',
					end: '2016-06-10'
				},
```

I will expect to see on the monthly view a event that cover 7,8,9 and 10 of the month.
By default if the time part is not specified the Allday is enabled.
So even 10th of the month must be covered and view in the event, instead is not like that:

![image](https://cloud.githubusercontent.com/assets/16102074/16844833/f16a0b26-49e6-11e6-9403-f61290778498.png)

Is possible to FIX it?
Thank you.

**Comment by arshaw** [comment]
event end dates are exclusive, this is documented here:
http://fullcalendar.io/docs/event_data/Event_Object/
(see `end` ... i wish this was better documented. apologies)

for further discussion on this, see:
https://github.com/fullcalendar/fullcalendar/issues/3190

**Comment by arshaw** [comment]
apologies for closing this. i thought i was commenting on a different issue. [this comment i previously made](https://github.com/fullcalendar/fullcalendar/issues/2985#issuecomment-187594947) expresses my view on the subject. we should change the way we parse.

**Comment by MikePayne** [comment]
@arshaw Has there been any progress on this issue?

**Comment by Ydalb** [comment]
@MikePayne : did you look at https://fullcalendar.io/docs/event_rendering/nextDayThreshold/ ?

**Comment by MikePayne** [comment]
@Ydalb Yes, but that does not affect allDay events. I've set the threshold to `00:00:00` (midnight) and I'm still seeing the behavior described above. I believe @arshaw's comment from last summer is still valid, just not implemented. 

>it's now a good idea to run the timed events through nextDayThreshold before converting them to ambiguously-timed moments for allDay events

**Comment by RajeshwariN** [comment]
@arshaw  
Hi, 
Can we split long event's horizontal bar into individual event and display it in full calendar?
